.github/workflows/reuseablehcpkgrelease.yaml

```diff
@@ -0,0 +1,39 @@
+name: compare branches
+on:
+  workflow_call:
+   inputs:
+    main:
+     description:  "Main Branch"
+     required: true
+     type: string
+    releasing:
+      description: "releasing Branch"
+      required: true
+      type: string
+
+
+jobs:
+  compare:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - run: | 
+          if [ "$(git rev-parse ${{inputs.releasing}})" != "$(git rev-parse ${{inputs.main}})" ]; then 
+          echo "Branches are different launching package release" 
+          echo '::set-output name=diff::launch'
+          else 
+          echo "Same"
+          exit
+          fi
+        id: hcsh 
+  launch:
+    needs: compare
+    if: ${{needs.hcsh.outputs.diff}} == 'launch'
```

Comment:
names can be improved for better readability imo, for example maybe we id the previous step "chkDiff" instead of "hcsh" and name the output var "decision" instead of "diff"